### PR TITLE
TF-2101 Show recover deleted message dialog

### PR DESCRIPTION
**Pull request by hieutbui**
## Issue
---
#2101 

### DoD:
- [x] Add `Recover deleted messages` option in context menu
- [x] Display recover deleted message dialog

### Need merge
---
[#2409 ](https://github.com/linagora/tmail-flutter/pull/2409)

## Resolved
---
- Web

https://github.com/linagora/tmail-flutter/assets/80142234/2f514c45-03a9-4e68-9650-61f6343944e8

- Android

https://github.com/linagora/tmail-flutter/assets/80142234/8ab133af-d8a9-4d8a-96d7-a0e6592c4818

### CODE REVIEW

File: lib/features/email_recovery/presentation/email_recovery_bindings.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  @override
+  void bindingsRepository() {
+    
+  }
+
+  @override
+  void bindingsRepositoryImpl() {
+    
+  }
+}
```
Comment (dab246) [inline lib/features/email_recovery/presentation/email_recovery_bindings.dart:45]: Override only necessary methods

**Review by hoangdat — APPROVED** [review]

File: lib/features/mailbox_dashboard/presentation/mailbox_dashboard_view_web.dart [hunk 2]
```diff
@@ -443,6 +444,25 @@ class MailboxDashBoardView extends BaseMailboxDashBoardView {
...
+              const SizedBox(width: 16),
+              TMailButtonWidget.fromIcon(
+                key: const Key('recover_deleted_messages_button'),
+                icon: controller.imagePaths.icRecoverDeletedMessages,
+                borderRadius: 10,
+                iconSize: 16,
+                onTapActionCallback: () => controller.gotoEmailRecovery(),
+              )
+            ],
+          );
```
Comment (dab246) [inline lib/features/mailbox_dashboard/presentation/mailbox_dashboard_view_web.dart:461]:
```suggestion
          return Padding(
            padding: EdgeInsetsDirectional.only(start: 16),
            child: TMailButtonWidget.fromIcon(
                key: const Key('recover_deleted_messages_button'),
                icon: controller.imagePaths.icRecoverDeletedMessages,
                borderRadius: 10,
                iconSize: 16,
                onTapActionCallback: () => controller.gotoEmailRecovery(),
              )
          );
```

File: lib/main/localizations/app_localizations.dart [hunk 3]
```diff
@@ -3660,4 +3660,88 @@ class AppLocalizations {
...
+  String get open {
+    return Intl.message(
+      'Open',
+      name: 'open',
+    );
+  }
+
+  String get bannerProgressingRecoveryMessage {
+    return Intl.message(
+      'The recovery is in progress. You can continue using TeamMail',
```
Comment (dab246) [inline lib/main/localizations/app_localizations.dart:3743]: Rename `TeamMail` to `TwakeMail`

**Comment by dab246** [comment]
@hieutbui  Please rebase it

**Review by dab246 — APPROVED** [review]

**Pull request merged**